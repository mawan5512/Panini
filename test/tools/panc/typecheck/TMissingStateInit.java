
/*
 * This file is part of the Panini project at Iowa State University.
 *
 * The contents of this file are subject to the Mozilla Public License
 * Version 1.1 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/.
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * For more details and the latest version of this code please see
 * http://paninij.org
 *
 * Contributor(s): Sean L. Mooney
 */

/*
 * @test
 * @summary Don't warn if state initialized in state modifier.
 * @compile/fail/ref=TMissingStateInit.out -Werror -XDrawDiagnostics TMissingStateInit.java
 */
capsule T () {
    int i;
    int k = 3;
    int j;
    int l = 4;

    => {
        i = 4;
        int j = 6;
        l = 10;
    }
}

