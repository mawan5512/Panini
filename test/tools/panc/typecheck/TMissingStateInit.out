TMissingStateInit.java:35:9: compiler.warn.state.already.initialized: l
TMissingStateInit.java:29:9: compiler.warn.state.not.initialized
- compiler.err.warnings.and.werror
1 error
2 warnings
